include/mega/types.h

```diff
@@ -671,6 +671,26 @@ namespace CodeCounter
     };
 }
 
+// Holds the config of a sync. Can be extended with future config options
+struct SyncConfig
+{
+    enum Type
+    {
+        TYPE_UP = 0x01, // sync up from local to remote
+        TYPE_DOWN = 0x02, // sync down from remote to local
+        TYPE_DEFAULT = TYPE_UP | TYPE_DOWN, // Two-way sync
+    };
+
+    // type of the sync, defaults to bidirectional
+    Type syncType = TYPE_DEFAULT;
+
+    // whether deletions are synced (only relevant for one-way-sync)
+    bool syncDeletions = false;
```

Comment:
wouldn't syncDeletions be true for two-way sync?  I it would probably keep the code simpler if we don't have to check the sync type before checking this flag?  And likewise for the one below?  thanks

Reply:
It is actually true for two-way. The function we'd use is `Sync::syncDeletions`. The config object should never be used directly when making sync decisions. I'll add some doc explaining that.